nautobot/extras/models/groups.py

```diff
@@ -354,6 +355,9 @@ def set_filter(self, form_data):
         # Use the auto-generated filterset form perform creation of the filter dictionary.
         filterset_form = filterset.form
 
+        # Get the declared form for any overloaded form field definitions.
+        declared_form = get_form_for_model(filterset._meta.model, form_prefix="Filter")
+
         # It's expected that the incoming data has already been cleaned by a form. This `is_valid()`
         # call is primarily to reduce the fields down to be able to work with the `cleaned_data` from the
         # filterset form, but will also catch errors in case a user-created dict is provided instead.
```

Comment:
Should we check `declared_form.is_valid` instead of `filterset_form.is_valid` below?

Reply:
We lean against form validation when filtering AFAICT, and ultimately the FilterSetForm should be doing the enforcement since it's what is based on the FilterSet.